File: example01databinding/src/main/java/foo/bar/example/foredatabinding/ui/wallet/WalletsActivity.java
package foo.bar.example.foredatabinding.ui.wallet;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentTransaction;

import foo.bar.example.foredatabinding.R;


public class WalletsActivity extends FragmentActivity {

    public static void start(Context context) {
        Intent intent = build(context);
        context.startActivity(intent);
    }

    public static Intent build(Context context) {
        Intent intent = new Intent(context, WalletsActivity.class);
        return intent;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.common_activity_base);

        if (savedInstanceState == null) {
            setFragment(
                    WalletsFragment.newInstance(),
                    WalletsFragment.class.getSimpleName());
        }
    }

    private void setFragment(Fragment fragment, String fragmentTag) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(
                R.id.content_main,
                fragment,
                fragmentTag);
        fragmentTransaction.commitAllowingStateLoss();
    }

}

File: app-examples/example-kt-01reactiveui/src/main/java/foo/bar/example/forereactiveuikt/ui/wallet/WalletsActivity.kt
package foo.bar.example.forereactiveuikt.ui.wallet

import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import co.early.fore.core.ui.SyncableView
import co.early.fore.kt.core.ui.LifecycleObserver
import foo.bar.example.forereactiveuikt.OG
import foo.bar.example.forereactiveuikt.databinding.ActivityWalletBinding
import foo.bar.example.forereactiveuikt.feature.wallet.Wallet

/**
 * Copyright © 2015-2020 early.co. All rights reserved.
 */
class WalletsActivity : FragmentActivity(), SyncableView {

    //models that we need to sync with
    private val wallet: Wallet = OG[Wallet::class.java]

    private lateinit var binding: ActivityWalletBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityWalletBinding.inflate(layoutInflater)
        setContentView(binding.root)

        lifecycle.addObserver(LifecycleObserver(this, wallet))

        setupButtonClickListeners()
    }

    private fun setupButtonClickListeners() {
        binding.walletIncreaseBtn.setOnClickListener {
            wallet.increaseMobileWallet() //notice how the reactive ui takes care of updating the view for you
        }
        binding.walletDecreaseBtn.setOnClickListener {
            wallet.decreaseMobileWallet() //notice how the reactive ui takes care of updating the view for you
        }
    }

    override fun syncView() {
        binding.walletIncreaseBtn.isEnabled = wallet.canIncrease()
        binding.walletDecreaseBtn.isEnabled = wallet.canDecrease()
        binding.walletMobileamountTxt.text = wallet.mobileWalletAmount.toString()
        binding.walletSavingsamountTxt.text = wallet.savingsWalletAmount.toString()
    }
}
